lib/chef/knife/list.rb

```diff
@@ -104,6 +104,18 @@ def add_dir_result(result)
         begin
           children = result.children.sort_by { |child| child.name }
         rescue Chef::ChefFS::FileSystem::NotFoundError => e
+          begin
+            rest.get("/clients/#{Chef::Config[:client_name]}")
+          rescue Net::HTTPServerException => e
+            if e.response.class == Net::HTTPNotFound
+              require "json"
+              begin
+                JSON.load(e.response.body())
```

Comment:
that should use ffi-yajl, and the require should get moved to the top of the file.

Reply:
This should happen in `chef/http` and not in individual classes.

Reply:
@thommay Did you mean ```chef/server_api```? I reckon that ```chef/http``` could be used somewhere else and this logic is specific to a chef server.